server/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala

```diff
@@ -14,67 +14,112 @@ import scala.util.{Failure, Success, Try}
 object ResourceService {
   private[this] val logger = getLogger
 
-  /** [[org.http4s.server.staticcontent.ResourceService]] configuration
-    *
-    * @param basePath prefix of the path files will be served from
-    * @param blocker execution context to use when collecting content
-    * @param pathPrefix prefix of the Uri that content will be served from
-    * @param bufferSize size hint of internal buffers to use when serving resources
-    * @param cacheStrategy strategy to use for caching purposes. Default to no caching.
-    * @param preferGzipped whether to serve pre-gzipped files (with extension ".gz") if they exist
-    */
-  final case class Config[F[_]](
+  class ResourceServiceBuilder[F[_]] private (
```

Comment:
I think we could move these to the top level.  Then we could leave the old interface as deprecated to assist people in migration.  What do you think?

Reply:
@rossabaker If this is going to go in 1.0, should we not just move away from the older interface right away? It's a breaking change anyway. Why keep the older interface?

Reply:
I like to keep them with `@deprecated` annotations, so it tells people how to migrate to the new interface.  People are a lot more likely to read their compiler warnings than they are the release notes.

Reply:
@rossabaker got it

Reply:
@rossabaker I have made the changes and pushed bulder to top level.